src/avram/upsert.cr

```diff
@@ -81,6 +81,27 @@ module Avram::Upsert
       yield operation, operation.record
     end
 
+    def self.bulk_upsert(upserts)
+      operations = upserts.map do |upsert_args|
+        new(**upsert_args)
+      end
+
+      upsert = Avram::BulkUpsert(self).new(
+        operations,
+        {{ attribute_names }}.to_a
```

Comment:
We're going to want a similar API, but something that hooks into database column constraints, since this isn't used for DB-level upsert currently.

Reply:
I believe this can be either a list of columns or a constraint name